.github/workflows/build.yaml

```diff
@@ -58,22 +58,23 @@ jobs:
       date: ${{ inputs.date }}
       sha: ${{ inputs.sha }}
   wheel-build:
-    runs-on: ubuntu-latest
-    container:
-      image: rapidsai/ci-conda:latest
-    defaults:
-      run:
-        shell: bash
-    steps:
-      - uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-      - name: Build wheel
-        run: ci/build_python_pypi.sh
-        env:
-          GH_TOKEN: ${{ github.token }}
-      - name: Publish distribution 📦 to PyPI
-        uses: pypa/gh-action-pypi-publish@release/v1
-        with:
-          password: ${{ secrets.RAPIDSAI_PYPI_TOKEN }}
-          skip-existing: true
+    secrets: inherit
+    uses: rapidsai/shared-workflows/.github/workflows/wheels-build.yaml@branch-24.02
+    with:
+      build_type: ${{ inputs.build_type || 'branch' }}
+      branch: ${{ inputs.branch }}
+      sha: ${{ inputs.sha }}
+      date: ${{ inputs.date }}
+      script: ci/build_wheel.sh
+      # Package is pure Python and only ever requires one build.
+      matrix_filter: map(select(.ARCH == "amd64" and .PY_VER == "3.10" and .CUDA_VER == "12.0.1"))
```

Comment:
@pentschev, you can generalize this with the following:

```yaml
matrix_filter: map(select(.ARCH=="amd64")) | sort_by(.CUDA_VER, (.PY_VER | split(".") | map(tonumber))) | .[-1]
```

That will filter the matrix to only include `amd64` machines, sort those machines by `CUDA_VER` and `PY_VER`, and then select the last element (which will have the latest CUDA and Python version).

You can optionally take this a step further to switch from a `centos7`/`rockylinux` image to an Ubuntu image with this minor amendment:

```yaml
matrix_filter: map(select(.ARCH=="amd64")) | sort_by(.CUDA_VER, (.PY_VER | split(".") | map(tonumber))) | .[-1] + {LINUX_VER: "ubuntu22.04"}
```

Though this will require the Ubuntu version to be updated periodically as RAPIDS rotates CI image versions.

Reply:
Also, if you don't care about the `CUDA_VER`, then this can be simplified to:

```yaml
matrix_filter: map(select(.ARCH=="amd64")) | sort_by((.PY_VER | split(".") | map(tonumber))) | .[-1]
```

and

```yaml
matrix_filter: map(select(.ARCH=="amd64")) | sort_by((.PY_VER | split(".") | map(tonumber))) | .[-1] + {LINUX_VER: "ubuntu22.04"}
``` 

Here's one final variation that can be used too, which sorts the entries by `PY_VER`, gets the last entry, and then overrides the `ARCH` and `LINUX_VER` values:

```yaml
matrix_filter: sort_by((.PY_VER | split(".") | map(tonumber))) | .[-1] + {ARCH: "amd64", LINUX_VER: "ubuntu22.04"}
```

Reply:
As Dask-CUDA is a pure Python package (agnostic to OS, Python version, CUDA version, etc.), think this is just selecting one build arbitrarily

Is the idea of these suggestions just to avoid hard-coding version information in the selector?

Reply:
Filed as issue: https://github.com/rapidsai/dask-cuda/issues/1296